pkg/backoff/backoff.go

```diff
@@ -0,0 +1,70 @@
+// Copyright 2018 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package backoff
+
+import (
+	"math"
+	"time"
+
+	"github.com/cilium/cilium/pkg/logging"
+	"github.com/cilium/cilium/pkg/uuid"
+
+	"github.com/sirupsen/logrus"
+)
+
+var log = logging.DefaultLogger
+
+// Exponential implements an exponential backoff
+type Exponential struct {
+	Min    time.Duration
```

Comment:
Can you add comments describing what is the purpose of each field?

Reply:
fixed